src/uct/api/tl.h

```diff
@@ -23,190 +23,342 @@ BEGIN_C_DECLS
+typedef void         (*uct_iface_progress_disable_func_t)(uct_iface_h iface,
+                                                          unsigned flags);
 
-    /* interface - tagged operations */
+typedef unsigned     (*uct_iface_progress_func_t)(uct_iface_h iface);
 
-    ucs_status_t (*iface_tag_recv_zcopy)(uct_iface_h iface, uct_tag_t tag,
-                                         uct_tag_t tag_mask,
-                                         const uct_iov_t *iov,
-                                         size_t iovcnt,
-                                         uct_tag_context_t *ctx);
+/* interface - events */
 
-    ucs_status_t (*iface_tag_recv_cancel)(uct_iface_h iface,
-                                          uct_tag_context_t *ctx,
-                                          int force);
+typedef ucs_status_t (*uct_iface_event_fd_get_func_t)(uct_iface_h iface,
+                                                      int *fd_p);
 
-    /* endpoint - pending queue */
+typedef ucs_status_t (*uct_iface_event_arm_func_t)(uct_iface_h iface,
+                                                   unsigned events);
 
-    ucs_status_t (*ep_pending_add)(uct_ep_h ep, uct_pending_req_t *n,
-                                   unsigned flags);
+/* interface - management */
 
-    void         (*ep_pending_purge)(uct_ep_h ep, uct_pending_purge_callback_t cb,
-                                     void *arg);
+typedef void         (*uct_iface_close_func_t)(uct_iface_h iface);
 
-    /* endpoint - synchronization */
+typedef ucs_status_t (*uct_iface_query_func_t)(uct_iface_h iface,
+                                               uct_iface_attr_t *iface_attr);
 
-    ucs_status_t (*ep_flush)(uct_ep_h ep, unsigned flags,
-                             uct_completion_t *comp);
+/* interface - connection establishment */
 
-    ucs_status_t (*ep_fence)(uct_ep_h ep, unsigned flags);
+typedef ucs_status_t (*uct_iface_get_device_address_func_t)(uct_iface_h iface,
+                                                            uct_device_addr_t *addr);
 
-    ucs_status_t (*ep_check)(uct_ep_h ep, unsigned flags, uct_completion_t *comp);
-
-    /* endpoint - connection establishment */
+typedef ucs_status_t (*uct_iface_get_address_func_t)(uct_iface_h iface,
+                                                     uct_iface_addr_t *addr);
 
-    ucs_status_t (*ep_create)(const uct_ep_params_t *params, uct_ep_h *ep_p);
+typedef int          (*uct_iface_is_reachable_func_t)(const uct_iface_h iface,
+                                                      const uct_device_addr_t *dev_addr,
+                                                      const uct_iface_addr_t *iface_addr);
 
-    ucs_status_t (*ep_disconnect)(uct_ep_h ep, unsigned flags);
 
-    void         (*ep_destroy)(uct_ep_h ep);
+/**
+ * Transport interface operations.
+ * Every operation exposed in the API should appear in the table below, to allow
```

Comment:
@hoopoepg can you pls rename 'should' -> 'must' as @tonycurtis noted, and squash all commits?

Reply:
@yosefe done